src/_reducers/assets.reducer.js

```diff
@@ -26,15 +29,43 @@ export default (state = initialState.assets, action) => {
         assetsList: [action.payload, ...state.assetsList],
         assetsCount: state.assetsCount + 1,
         hasError: false,
-        isLoading: false
+        isLoading: false,
+        success: 'Hoooray! Asset successfully created. You can create another one or head on to view all assets.',
+        errorMessage: ''
       };
 
-    case CREATE_ASSET_FAIL:
+    case CREATE_ASSET_FAIL: {
+      let message;
+      const { data } = action.payload.response;
+
+      if (data.hasOwnProperty('asset_code')) {
```

Comment:
Move the if's into a function just before the reducer function. Don't bother about creating a file for it. I.e.
```js
// Add a comment why we are retrieving the error messages as such, including the format of what the API currently returns
// Also, add a TODO comment saying this will be updated once the API returns the error message in a better format
const getErrorMessage = error => {
  if (error.hasOwnProperty('asset_code')) {
    return error.asset_code[0].replace(/[[\]']/g, '');
  }
  if (error.hasOwnProperty('serial_number')) {
    return error.serial_number[0].replace(/[[\]']/g, '');
  }
  if (error.hasOwnProperty('non_field_errors')) {
    return error.non_field_errors[0];
  }
  return '';
}

export default (state = initialState.assets, action) => {
  switch(action.type) {
    ...
    case CREATE_ASSET_FAIL: 
      return {
        ...state,
        hasError: true,
        isLoading: false,
        success: '',
        errorMessage:  getErrorMessage(action.payload.response.data)
      }
  }
}